chalice/app.py

```diff
@@ -379,6 +429,20 @@ def _register_view(view_func):
             return view_func
         return _register_view
 
+    def _validate_route_content_types(self, name, content_types):
+        # A routes content_types be homogeneous in their binary support.
+        binary, non_binary = [], []
+        for content_type in content_types:
+            if content_type in self.api.binary_types:
+                binary.append(content_type)
+            else:
+                non_binary.append(content_type)
+        if binary and non_binary:
+            raise ValueError(
+                'In view function "%s", the content_types %s support binary '
+                'and %s do not. All content_types must be consistent in their '
```

Comment:
`raw_body` will be `bytes` if it was triggered by a binary `Content-Type` under py3. 

Lets say my api is for resizing images. So I have something like this:
```python
@app.route(..., content_types=['image/tff', 'image/png', ...])
def foo():
    resized = _resize_image_and_make_png(app.current_request.raw_body)
    return Response(status_code=200,
                 body=resized,
                 headers={'Content-Type': 'image/png'})
```

I may think everything is fine for a long time because tff isn't super common and maybe I don't do an integration test using that format, and when I send a tiff up it will error out. Because I am passing a string to the resized function. I may even have trouble debugging this issue because inspecting this function, everything appears to be fine. It's not until my assumption that `image/tff` is treated as a binary type is broken that I will even look for documentation regarding what is and is not a binary type. I may never have even looked at the docs regarding this in the past since binary will 'just work' for the most part. 

This error message prevents this entire class of fault, and calls out which types are mismatched. With this information I will be prompted to go look at the documentation regarding which types aren't binary, and will be lead to the docs regarding `app.api.binary_types`. With that information I can align the content-types I accept with the binary_types.